src/plugins/intel_cpu/CMakeLists.txt

```diff
@@ -22,7 +22,8 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
     # oneDNN arm64: unary minus operator applied to unsigned type, result still unsigned
     ov_add_compiler_flags(/wd4146)
 elseif(OV_COMPILER_IS_CLANG)
-    ov_add_compiler_flags(-Wno-delete-non-abstract-non-virtual-dtor)
+    # -Wno-delete-non-abstract-non-virtual-dtor is not fully supported by clang
+    ov_add_compiler_flags(-Wno-delete-non-virtual-dtor)
```

Comment:
maybe we need to select proper option based on clang version?

Reply:
Good idea, I check the clang document `-Wno-delete-non-abstract-non-virtual-dtor` is support after [clang-8](https://releases.llvm.org/8.0.0/tools/clang/docs/DiagnosticsReference.html)